### PR TITLE
added project carbon_rover

### PR DESCRIPTION
**Pull request by ric96**
- added source files for project
- added README.md for the project

Signed-off-by: Sahaj Sarup <sahaj.sarup@linaro.org>

### CODE REVIEW

File: carbon_rover/README.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+# 1) Hardware
+
+## 1.1) Hardware requirements
+
+- [96Boards Ccarbon IE](https://www.96boards.org/product/carbon/)
+- 5v 2A USB Power Bank
+- [6x Ultrasonic Sensors HC-SR04](https://www.robomart.com/buy-4-pin-ultrasonic-sensor-module-arduino-raspberry?search=hcsr04&dsearch=hcsr04)
+- 6x 1KOhm Resistors
+- 6x 2KOhm Resistors
+- [4x IR Edge Detection Sensors](https://www.robomart.com/ir-based-digital-color-sensor-black-white)
```
Comment (Mani-Sadhasivam) [inline carbon_rover/README.md:30]: IR sensor quantity is 4 or 6?

File: carbon_rover/rover/src/main.c [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+void fwd();
+void bwd();
+void stop();
+void left();
+void right();
+uint32_t get_us(uint32_t trig, uint32_t echo, struct device *dev);
+
+void main(void)
+{
+	gpioa = device_get_binding(PORT);
```
Comment (Mani-Sadhasivam) [inline carbon_rover/rover/src/main.c]: Add error check please

**Review by Mani-Sadhasivam — CHANGES_REQUESTED** [review]
Added trivial comments. Overall LGTM

File: carbon_rover/README.md (hunk 1 above)
Comment (ric96) [reply carbon_rover/README.md:30]: 4, one for each corner

File: carbon_rover/rover/src/main.c [hunk 3]
```diff
@@ -60,36 +60,107 @@ uint32_t get_us(uint32_t trig, uint32_t echo, struct device *dev);
 
 void main(void)
 {
+	int ret;
+
 	gpioa = device_get_binding(PORT);
+	if (!gpioa) {
+		printk("Cannot find %s!\n", PORT);
```
Comment (Mani-Sadhasivam) [inline carbon_rover/rover/src/main.c]: We should return here. No point in continuing once the port is not found.

**Review by Mani-Sadhasivam — CHANGES_REQUESTED** [review]

**Review by Mani-Sadhasivam — APPROVED** [review]

**Pull request merged**